# Notes on the introduction papers to this course

At some point, a teacher said "these papers roughly outline the course content"~.

The 3rd task of today: "read up on yesterday’s general genome variation papers & make notes / think about questions for discussion rounds".

Papers 1-4 are yet to be read. Notes regarding the task will be added in this file.

Four _general genome variation papers_ introducing the topic of this course:

## 1. [Finding the Rare Pathogenic Variants in a Human Genome]https://internal.baudisgroup.org/pdf/articles/2017-05-09___Evans__Clinical_genome_sequencing__JAMA_comment.pdf
(2 pages) Copyright 2017 American Medical Association. All rights reserved.

### Vocabulary ([deepl.com](https://www.deepl.com/translator#en/de/concomitantly%0Aameliorate) ):
- concomitantly: begleitend
- ameliorate: verbessern

### Summary of paper:
Reference sequences are typically a composite of sequences from many different genomes. Sequences of special interest in unhealthy individuals' genomes variant analysis are e.g. sense mutations.
Certain limitations - "current understanding of most genetic variation in humans is inadequate", "the interpretation and implications of genomic findings fundamentally differ in healthy individuals vs those with disease" - make routine genomic sequencing of healthy individuals inadvisable (paper published 2017-05-09).
Investigating not whole genomes of (un)healthy individuals but rather specific genome regions known to be critical  might prove more effective and efficient in gaining useful, i.e. actionable, knowledge about genetically caused diseases.

### Discussion ideas, interests in topic:
- It seems to me that the problem of the certainty with which variants are associated with disease could be connected to and obscure by the lack of knowledge about disease penetrance in an individual. For instance, DNA repair of pathogenic variants before development of disease could be something reducing the penetrance of possible pathogenic variants, I think.
-- This idea was written down before reading the section "Evidence Supporting Genome Sequencing in Healthy Individual". Thus, the idea of investigating penetrance as a more limiting issue is somewhat supported by this paper.
- Although getting cheaper, gathering genomic data harbors data privacy issues and has many unpleasant implications, imo. (elaborate)

## 2. https://internal.baudisgroup.org/pdf/articles/2016-01-15___Heather_and_Chain__The-sequence-of-sequencers__Genomics.pdf 
(8 pages)

_'First-generation' sequencing method(s)_:
`note to self: cut down this section into short descriptions of: 1st gen seq (Sanger); shotgun seq;`

- Sanger: see other local repo. aka 'plus and minus' system. DNA polymerase is used. 'plus reaction': only have one nucleotide available - all strands end with that nucleotide. 'minus reaction': only have the other three nucleotides - all strands are elongated up to the point where the fourth nucleotide would follow.
Used polyacrylamide to resolve DNA fragments.
- Maxam-Gilbert: aka chemical cleavage technique.
> ... radiolabelled DNA is treated with chemicals which break the chain at specific bases; after running on a polyacrylamide gel the length of cleaved fragments (and thus position of specific nucle- otides) can be determined and therefore sequence inferred (see Fig. 1, right). This was the first technique to be widely adopted, and thus _might be considered the real birth of ‘first-generation’DNA sequencing_.
Also used polyacrylamide to resolve DNA fragments.

_The next significant improvement_:
-  1977 Sanger's ‘chain-termination’or dideoxy technique, aka **'Sanger method'**:
  - p.2: "... strands of each possible length being produced, as the dideoxy nucleotides get randomly incorporated as the strand extends, halting further progression."
  - 

## 3. uhttps://internal.baudisgroup.org/pdf/articles/2015-10-01___1000-Genomes-Consortium__A-global-reference-for-human-genetic-variation__Nature.pdf
(20 pages)

## 4. https://internal.baudisgroup.org/pdf/articles/2011-05-12___Eichler_et_al__Review_structural_variations_arrays__NatRevGen.pdf
(14 pages)
