
# Genome technologies
## First generation sequencing

### Sanger Sequencing
* The first-generation sequencing method is the Sanger sequencing method developed in 1977 and still used nowadays where no high throughput is necessary. The Sanger method uses fluorescent dye-terminator reagents for single-reaction sequencing.

---
## Second generation sequencing

#### Sequencing by hybridization
-  This sequencing method was developed in 1980 and is using an arrayed DNA oligonucleotide of sequence placed on filters. Those will hybridize to known labelled fragment of the DNA, which needs to be sequenced. With a repetition of hybridization and washing steps, it is then possible to check which labelled fragment is hybridized to the matching sequence of DNA probes. 

#### Sequencing by synthesis: 
-  This method can be run without dideoxy terminators and rely on shorter reads compared to the Sanger Seqeunceing method) (300-500bases).

>454 Pyrosequencing: 
- This method detects the light produced by the release of pyrophosphate once a nucleotide is correctly incorporated in a growing DNA chain. 
  Pyrosequencing was used for genome sequencing and Metagenome samples due to the generation of long read length, 99% accuracy and high throughput (25million bases). However, nowadays this method was discontinued by Roche maybe due to impactful disadvantages such as a high cost of reagents and a high error rate in homopolymer repeats.

>Ion torrent:

- Ion torrent is using an ion sensor to record the voltage change induced by the released of a hydrogen ion. The hydrogen ion will be released when a nucleotide is correctly incorporated across its complementary base in a growing DNA chain. Moreover, the sequencing reaction will be conducted in well containing a microchip, which will be flushed with individual unlabelled dNTPS. and electronic sensors located at the bottom of each cell. Advantages are the flexible workflow, the longer read length and cheaper price. One disadvantage is the high rate of sequencing errors


>Illumina:

- Illumina will detect fluorescently labelled chain terminating nucleotides. The generation of the DNA template for sequencing will be conducted using the “PCR bridge amplification” forming small colonies. Advantages of using Illumina are the high output of sequencing per run (600GB), the simplicity, short read length (100bp) and the cost.

---


## Third generation sequencing
* Third generation sequencing methods aim to sequence long DNA/RNA sequence

>PacBio Bioscience (PacBio):

- PacBio also known as Single Molecule Real Time (SMRT) sequencing. It is performed in SMRT cell containing ultra-microwells (zeptoliter scale), where one molecule of DNA polymerase is fixed on the bottom of each well. Fluorescently labelled dNTPs are added on the template single strand DNA, which is linked to the DNA polymerase: If one dNTP is incorporated, the fluorescence of those dNTPs can be detected. Advantages are, th identification of methylation sites, long reads, which is good for genome assemblies and the rapid measurement rate of added nucleotides. One disadvantage is that it has a high error rate

---

## Fourth generation sequencing

> Oxford Nanopore Technology:

It is a single molecule sequencing system where the conductivity of ion current in the nanopore will changes once a strand of nucleic acid go through this pore. Advantages of the nanopore method is that it is simple, do not require long sample preparation, no need of either ligation nor amplification before sequencing and long reads can be obtained. One disadvantage is that the speed of the translocation of DNA through the nanopore should be improved to ensure a reliable measurement.

---

## Cytogenetic techniques:

> M-Fish method:

M-Fish (multi-fluor Fish) is a method to study marker chromosomes and complex chromosomal aberrations/re such as tumours. Initially it was developed to stain human chromosomes, however M-Fish has been extended to other species (e.g. mouse). The 24 colour M-Fish protocol allows to label region-specific probes with a unique combination of different fluorochromes allowing a unique emission spectrum for each probe. Moreover, it is important to use the appropriate band-pass fluorescence filters and sophisticated digital imaging software to yield a good karyotyping image.


