# Genome variation papers overview

### Finding the Rare Pathogenic Variants in a Human Genome
An often proposed idea is to perform wgs on healthy humans as a regular procedure to detect possible pathogenic variants and be able to early respond with the proper treatment 
(before pathogene has severe consequences). However appealing this idea might be, at the current stage it is premature and inadvisable to carry it out.
The main caveats are:
*  current understanding of most genetic variation in humans is inadequate
*  so far variant analysis was often carried out on sick patients and thus:
   *  the certainty with which a variant is associated with disease is often overestimated (false-positives)
   *  the penetrance of the condition (likelihood that disease will develop) is often overestimated (overdiagnoses)
* even though sequence cost has plummeted, substantial medical costs arise from interpretating the resulting data
  
> Momentarily a better strategy is screening healthy humans on a small number of well-understood human genes, which mutated, result in a high risk of readily preventable disease.

<br/>

### The sequence of sequencers: The history of sequencing DNA
> Determining the order of nucleic acid residues in biological samples is an integral component of a wide variety of research applications.  

In the last 50 years there has been huge progress in the fiel of sequencing. Starting struggling with sequencing a single gene to rapid and widely available wgs nowadays.  
#### First-generation DNA sequencing  
* Able to measure nucleotide composition
* Sanger sequencing:  detection of radiolabelled partial-digestion fragments after two-dimensional fractionation.  (sequence-by-synthesis)

#### Second-generation DNA sequencing (NGS)  
* Use of luminescent method for measuring pyrophosphate synthesis
* mass parallelisation of sequencing reactions (hundreds of basepairs in length)
* high-throughput sequencing (HTS)
* Illumina etc
#### Third-generation DNA sequencing  
* There is scientific discussion on what exactly criteria for 3. gen sequencing are:
  * single molecule sequencing (SMS)
  * real-time sequencing
  * negating the requirement for DNA amplification shared by all previous technologies
  * divergence from previous technologies
* eg. SMRT of PacBio

<br/>

### A global reference for human genetic variation
> The 1000 Genomes Project set out to provide a comprehensive description of common human genetic variation by
applying whole-genome sequencing to a diverse set of individuals from multiple populations. Here we report
completion of the project, having reconstructed the genomes of 2,504 individuals from 26 populations using a combina-
tion of low-coverage whole-genome sequencing, deep exome sequencing, and dense microarray genotyping.  

Ability to accurately capture structural variation remains more limited

<br/>

### Genome structural variation discovery and genotyping
> Comparisons of human genomes show that more base pairs are altered as a result of structural variation — including copy number variation — than as a result of point mutations. 
  
* NGS has lead to exponential increase in the discovery of smaller structural-variation events
* however some global discovery biases remain (some ethnicities better covered then others)
* HQ de novo assembly of human genomes needed to get full picture of all classes of structural variation (beyond reach of current tech, parallel sequencing of long fragments needed for reasonable cost)

**Copy number variation (CNV)**:  
Phenomenon in which sections of the genome are **repeated and the number of repeats in the genome varies between individuals.**  
Copy number variation is a type of structural variation: specifically, it is a type of duplication or deletion event that affects a considerable number of base pairs.  
Approximately two-thirds of the entire human genome may be composed of repeats[3] and **4.8–9.5% of the human genome can be classified as copy number variations**.  
In mammals, copy number variations play an **important role in generating necessary variation in the population** as well as disease phenotype. (Wikipedia)
















