# Genome resources

## Overview

### Task description
* Explore genome resources and provide some notes in a doc posted on Github

### Some genome resources
* **Reference genomes**
  * [UCSC Genome Browser](http://genome.ucsc.edu/)
  * [Human Genome Resources at NCBI](https://www.ncbi.nlm.nih.gov/projects/genome/guide/human/)
  * [Ensembl](https://www.ensembl.org/index.html)
* **Genome variant data**
  * [NCBI:dbSNP](https://www.ncbi.nlm.nih.gov/snp/)
  * [NCBI:dbVAR](https://www.ncbi.nlm.nih.gov/dbvar/)
  * [NCBI:ClinVar](https://www.ncbi.nlm.nih.gov/clinvar/)
  * [EMBL-EBI:EVA](https://www.ebi.ac.uk/eva/)
  * [IGSR: The International Genome Sample Resource](https://www.internationalgenome.org/)
* **Cancer genomics**
  * [COSMIC](https://cancer.sanger.ac.uk/cosmic)
  * [ClinGen](www.clinicalgenome.org)
  * [ICGC Cancer Genome Projects](https://dcc.icgc.org/)
  * [PMKB Precision Medicine Knowledgebase](https://pmkb.weill.cornell.edu)
  * [Progenetix arrayMap](https://arraymap.progenetix.org/progenetix-cohorts/arraymap/)

## Reference genomes

### UCSC Genome Browser
* Seems well curated (last update on Sep. 13, 2021)
* Many different tools with specific functions
* Different species
* Layout quite old-fashioned
* Genome browser not really intuitive (or I'm not proficient enough)
* Tons of information about the different genes

### Human Genome Resources at NCBI
* I like the little window showing the chromosomes
* Tons of different species, from soybeans to fruit flies
* Different human variant collections
* Many different tracks can be shown (as in the UCSC one)

### Ensembl
* Tons of different species, including Mummichogs (no idea what that is)
* Regular updates every two weeks
* Also RNA seq data
* I like the layout of this one better, it's less crowded and more structured
* They tricked me into thinking they had some histology pictures, but it was just the picture for the gene expression

## Genome variant data

### NCBI:dbSNP
* *dbSNP contains human single nucleotide variations, microsatellites, and small-scale insertions and deletions along with publication, population frequency, molecular consequence, and genomic and RefSeq mapping information for both common variations and clinical mutations.*
* You can look for gene names or IDs, genomic location on chromosomes and for clinical significance
* Additional filter function

### NCBI:dbVar
* *dnVar is NCBI's database of human genomic Structural Variation - large variants > 50 bp including insertions, deletions, duplications, inversions, mobile elements, translocations, and complex variants.*
* Quite similar to dbSNP
* You can look for data from specific studies
* References are listed when you click on a variant

### NCBI:ClinVar
* *ClinVar aggregates information about genomic variation and its relationship to human health.*
* You can look for conditions and filter according to clinical significance, variation type and more
* Different review statuses and method types
* Nice layout when you click on one of the variants, easy to understand

### EMBL-EBI:EVA
* *The European Variation Archive is an open-access database of all types of genetic variation data from all species.*
* I like the statistics on the front page
* Seems to be regularily updated
* Links to Ensembl
* If I understand it correctly, you can look for specific studies and the position on the chromosome, but not for gene names

### IGSR: The International Genome Sample Resource
* *The International Genome Sample Resource (IGSR) maintains and shares the human genetic variation resources built by the 1000 Genomes Project. We also update the resources to the current reference assembly, add new data sets generated from the 1000 Genomes Project samples and add data from projects working with other openly consented samples.*
* Last announcement from August 2020
* Goals:
  * Ensure the future access to and usability of the 1000 Genomes reference data
  * Incorporate additional published genomic data on the 1000 Genomes samples
  * Expand the data collection to include new populations not represented in the 1000 Genomes Project
* Works with *Ensemble*

## Cancer genomics

### COSMIC
* You can browse by tissue type and histology, and there are a LOT of options
* If you search for a condition, you get the top 20 genes, genes with mutations and genes without mutations that are correlated
* You can look for different kinds of variants
* Quite easy to understand genome browser

### ClinGen
* Regularily updated
* Really nice layout
* Interesting options like Gene-Disease Validity, Variant Pathogenicity, Clinical Actionability, Dosage Sensitivity, Somatic Cancer Variant and Baseline Annotation
* Seems to be mostly correlations(?)

### ICGC Cancer Genome Projects
* Also really nice layout
* Quite a few external references and also an overview over mutations and drugs for specific genes
* Nice interface to look through the cancer projects
* Apparently the ICGC [ARGO Data Platform](https://platform.icgc-argo.org/) is the successor(?)

### PMKB Precision Medicine Knowledgebase
* You can check different mutations in cancers of different regions of the body, with a short explanation and clinical relevance
* You can also check different tumors and mutations that are present in them
* Easy to understand

### Progenetix arrayMap
* Data from different sources
* Nice search function
* There's a plotting tool which sounds pretty cool

