# Genome resources

## Overview

### Task description
* Explore genome resources and provide some notes in a doc posted on Github

### Some genome resources
* **Reference genomes**
  * [UCSC Genome Browser](http://genome.ucsc.edu/)
  * [Human Genome Resources at NCBI](https://www.ncbi.nlm.nih.gov/projects/genome/guide/human/)
  * [Ensembl](https://www.ensembl.org/index.html)
* **Genome variant data**
  * [NCBI:dbSNP](https://www.ncbi.nlm.nih.gov/snp/)
  * [NCBI:dbVAR](https://www.ncbi.nlm.nih.gov/dbvar/)
  * [NCBI:ClinVar](https://www.ncbi.nlm.nih.gov/clinvar/)
  * [EMBL-EBI:EVA](https://www.ebi.ac.uk/eva/)
* **Cancer genomics**
  * [COSMIC](https://cancer.sanger.ac.uk/cosmic)
  * [ClinGen](www.clinicalgenome.org)
  * ICGC Cancer Genome Projects
  * [PMKB Precision Medicine Knowledgebase](https://pmkb.weill.cornell.edu)
  * arrayMap
  * Progenetix Cancer CNV Information Resource

## Reference genomes

### UCSC Genome Browser
* Seems well curated (last update on Sep. 13, 2021)
* Many different tools with specific functions
* Different species
* Layout quite old-fashioned
* Genome browser not really intuitive (or I'm not proficient enough)
* Tons of information about the different genes

### Human Genome Resources at NCBI
* I like the little window showing the chromosomes
* Tons of different species, from soybeans to fruit flies
* Different human variant collections
* Many different tracks can be shown (as in the UCSC one)

### Ensembl
* Tons of different species, including Mummichogs (no idea what that is)
* Regular updates every two weeks
* Also RNA seq data
* I like the layout of this one better, it's less crowded and more structured
* They tricked me into thinking they had some histology pictures, but it was just the picture for the gene expression

## Genome variant data

### NCBI:dbSNP
* *dbSNP contains human single nucleotide variations, microsatellites, and small-scale insertions and deletions along with publication, population frequency, molecular consequence, and genomic and RefSeq mapping information for both common variations and clinical mutations.*
* You can look for gene names or IDs, genomic location on chromosomes and for clinical significance
* Additional filter function

### NCBI:dbVar
* *dnVar is NCBI's database of human genomic Structural Variation - large variants > 50 bp including insertions, deletions, duplications, inversions, mobile elements, translocations, and complex variants.*
* Quite similar to dbSNP
* You can look for data from specific studies
* References are listed when you click on a variant

### NCBI:ClinVar
* *ClinVar aggregates information about genomic variation and its relationship to human health.*
* You can look for conditions and filter according to clinical significance, variation type and more
* Different review statuses and method types
* Nice layout when you click on one of the variants, easy to understand

### EMBL-EBI:EVA
* *The European Variation Archive is an open-access database of all types of genetic variation data from all species.*
* I like the statistics on the front page
* Seems to be regularily updated
* Links to Ensembl
* If I understand it correctly, you can look for specific studies and the position on the chromosome, but not for gene names

## Cancer genomics

### COSMIC
* You can browse by tissue type and histology, and there are a LOT of options
* If you search for a condition, you get the top 20 genes, genes with mutations and genes without mutations that are correlated
* You can look for different kinds of variants
* Quite easy to understand genome browser
