# Terminal and Unix 

## FastQ Format 
Fasta with Qalities is a plain text file. 
It contains 4 lines per read sequence: 
* _line 1_: header line, identification of the sequence "@", sequencing technique  
* _line 2_: read sequence 
* _line 3_: comment line "+"
* _line 4_: Phred quality score (i.e. "a measure of the quality of the identification of the nucleobases generated by automated DNA sequencing")

> **Phred Quality Score**: Q = −10 log10 P (1 in 10%, 1 in 100, etc) and is described by letters. Different depending on the sequencing used!!! Phred+33, Solex+64, Phred+64. For Illumina 1.8+, Phred+64, which matches with the original one, Phred+33. 

**Useful command lines**: 
* To determine the number of sequence (number of line / 4): 
`wc file.fq | awk '{print $1 / 4}'`
* To retrieve each 1st/2nd/3rd/4th line of the read sequence (i.e. the sequence): 
`awk 'NR%4==1' file.sq` (NR: current line number; % modulo) (==2 / ==3/ ==0)
* To convert FastQ to Fast (and save): 
```
awk 'NR % 4 == 1 {print ">"$1}; 
     NR % 4 == 2 {print}' SP1.fq \
     | > example.fa
```
* To count the number of reads:
`awk 'END{print NR/4}' file.fq`
     
## BED Format 
"BED (Browser Extensible Data) format provides a flexible way to define the data lines that are displayed in an annotation track". BED files consist at least of three fields: 
* **chrom**: the chromosome's name
* **chromStart**: the starting position 
* **chromEnd**: the ending position 
> and nine additional optional fields (name, score, strand, thickStart, thickEnd, itemRgb, blockCount, blockSizes, blockStarts)

**Useful command lines**: 
* To transform a BED6 to BED3 file (extract the first three columns): \
`awk -v OFS='\t' '{print $1, $2, $3}' file.bed`
* to add a nucleotide to the start and substract a nucleotide to the end: \
`awk -v OFS='\t' '{print $1,$2-1,$3-1,$4,$5,$6}' file.bed`
* to intersect two BED files (check for overlapping): \
```
alias bedtools='~/course/soft/bedtools2/bin/bedtools'
bedtools intersect \
  -file1  ~/path/to/the/directory/file1.bed \
  -file2  ~/path/to/the/directory/file2.bed  
```
* to report intervals from a BED file 1 that do not overlap any of the intervals in BED file 2: \
```
bedtools intersect \
  -v \
  -file1  ~/path/to/the/directory/file1.bed \
  -file2  ~/path/to/the/directory/file2.bed  
```
* to flag the amounts of overlap for all features when comparing two BED files: \
```
bedtools intersect \
  -wao \
  -file1  ~/path/to/the/directory/file1.bed \
  -file2  ~/path/to/the/directory/file2.bed  
```

> FASTAQ -> SAM (aligned against reference genome) -> BED (counts)
> BED (BED12) format is typically used for **genome browswer**. 

## GFF3 & GTF File Format 
General Feature Format (GFF) is a tab-delimited text file that contains information about genomic as well as protein sequences feature. General Transfer Format (GTF) is identical to GFF. 

GFF: 
```
X	Ensembl	Repeat	2419108	2419128	42	.	.	hid=trf; hstart=1; hend=21
X	Ensembl	Repeat	2419108	2419410	2502	-	.	hid=AluSx; hstart=1; hend=303
X	Ensembl	Repeat	2419108	2419128	0	.	.	hid=dust; hstart=2419108; hend=2419128
X	Ensembl	Pred.trans.	2416676	2418760	450.19	-	2	genscan=GENSCAN00000019335
X	Ensembl	Variation	2413425	2413425	.	+	.	
X	Ensembl	Variation	2413805	2413805	.	+	.

```

GTF: 
```
1 transcribed_unprocessed_pseudogene  gene        11869 14409 . + . gene_id "ENSG00000223972"; gene_name "DDX11L1"; gene_source "havana"; gene_biotype "transcribed_unprocessed_pseudogene"; 
1 processed_transcript                transcript  11869 14409 . + . gene_id "ENSG00000223972"; transcript_id "ENST00000456328"; gene_name "DDX11L1"; gene_sourc e "havana"; gene_biotype "transcribed_unprocessed_pseudogene"; transcript_name "DDX11L1-002"; transcript_source "havana";
```

> Perfect to store **genomic annotations**! 
  
  **Useful command lines**: 
  * to retrieve the details of a specific transcripts (e.g. `ENST00000342247`) from a GTF file: \
 ` grep ENST00000342247 file.gtf | grep "exon\s" ` \
 and the number of exons: \
 ` grep ENST00000342247 file.gtf | grep "exon\s" | wc -l `
 * to determine the number of start / stop codon the chromosome have : \
 ` grep start_codon file.gtf | wc -l ` replace by stop_codon
 
## BEDgraphs File Format
BEDgraphs format displays continuous-valued data in track format. 

> Useful for **probability scores** and **transcriptome data**. 

## Wig File Format
Wig format is similar to BEDgraphs, i.e. displays continuous-valued data in track format. Exercise: " Improved Wig: how to represent the scores of 300 nucleotides from chr3, starting from 400601, where the 􏰥rst 100 nt are scored 11, the next 100 nt 22, and the last 33?" 

```
fixedStep chrom=chr3 start=400601 step=100 span=5
11
22
33
```

> Useful for **GC percent, probability scores** and **transcriptome data**. 

  
     
## Swiss-Prot flatfile 
Protein entries. Each line starts with a two character line code such as ID for the first, followed by "w\". The entry is ended by "\\". 

## Unix shell 

  Command         | Function        
| --------------- |:----------------|
| history         |history of the commands used in the shell |
| clear           | clear the CLI   |
| kill            | terminate or signal a process |
|*                | matches any number of characters (none, one or more) in a file or directory name|
|? | matches exactly one character - use '??' to match any group of 2 characters|
| [] | specify a range of characters allowed at that position - use '!' to exclude a range of characters at that position |
| {} | specify a list of terms - separated by commas |
| ^ | All the lines that start with | 
| $ | All the lines that end with | 
| mv | move or rename a file or a directory | 
| cp _file, new name, location_ | Copy a file or a directory |
| tar | manipulate archives |
| chmod | Change file modes or access control lists | 
| find | walk a file hierarchy: _-name 'FileName'_ |
| cat | display file content |
| less | display the content of a text file one screen at a time (text viewer)|
| sort | sort or merge records (lines) of text and binary files (-u (unique), -n (numeric), -r (reverse))|
| \r | newline with MacOS, corresponding to '^M', UNIX '\n' |
| tr | translate characters, -d (delete), '' '' |
| diff | compares two files and retrieve the difference |
| grep | searches file(s) for words or patterns and displays the result lines | 
| uniq | report one line if several identical and adjacent lines are found | 
| gunzip | uncompress .gz files | 


**Pipe "|"** allows us to pass from the output from one program **directly** to the input of another program.

