---
title: "CNV of human Glioblastoma"
author: "Daniel Walther"
date: "10/01/2021"
output: pdf_document
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Step 1: Install package

```{r}
if (!require(devtools)){
    install.packages('devtools')
}
devtools::install_github('progenetix/pgxRpi')
library(pgxRpi)
```

## Step2: Search Glioblastoma NCIt code (see pdf on course page)

```{r}
ncit_code <- "NCIT:C3058"
```

## Step3: Access the CNV frequency data from samples with Glioblastoma

```{r}
freq <- pgxLoader(type='frequency', output='pgxseg',filters=ncit_code,
                  codematches=T)
```

### The retreived data is an object contaning two slots `meta` and `data`.

The `meta` slot looks like this (contains metadata, like sample count):

```{r}
freq$meta
```

The `data` slot includes two matrices.

```{r}
names(freq$data)
```

The columns gain_frequency and loss_frequency are of primary interest.

```{r}
head(freq$data$`NCIT:C3058`)
```

```{r}
dim(freq$data$`NCIT:C3058`)
```

## Step4: Visualize data

### By genome

```{r}
pgxFreqplot(freq)
```

### By chromosome

```{r}
par(mfrow = c(2, 2))
pgxFreqplot(freq, chrom = 9)
pgxFreqplot(freq, chrom = 13)
pgxFreqplot(freq, chrom = 14)
pgxFreqplot(freq, chrom = 22)
```

## Step5: Analyse the data

According the plot, we can see frequency gains on chromosome 7p, 7q, 19p, 19q, 20p, 20q.
and frequency losses on chromosome 9p, 10p, 10q, 13q, 14q, 15q, 17p, 18q, 22q.

The threshold for mentioning was arbitrarily chosen at roughly 20% gain or loss respectively (based on what sticks out in the genome visualization).

On chromosomes 13p, 14p and 22p there is nearly no gain at all and maybe 1% loss. This could indicate deletion of the whole p-half (potentially even both parental copies) of these chromosomes.

It would be interesting formulate a profile of gain versus loss CNV with respect to locus (in varying detail, e.g. which genes affected) and to compare this profile with other forms of malignant tumours.

## Step6: Compare to peer studies

The proper workflow would be at this point to research literature regarding this topic to find supporting and/or contradicting result.
