# [A copy number variation map of the human genome](https://www.nature.com/articles/nrg3871)
##### Exercises

#### Q1: Why is important to construct a CNV map on health individuals of various ethnicities? (Introduction)
There is a continous spectrum of phenotypic effects of CNV from adaptive traits to embryonic lethality. The threshold of which is defined by clinicans and are arbetrary
But traits can e adaptive and maladaptive in different envionmental contexts such as diet.

#### Q2: What is the CNV size that the authors defined? (Box 1 mentioned in introduction)
50 bp to 3Mb

#### Q3: What are the primary approaches used for CNV detection? And what are the advantages and limitations 
for these technologies? (CNV discoveries) 

|technique|advantage|limitation|
|---------|---------|----------|
|arrays based| study quantitative variants| inflation ==> low resolution and overestimation of breakpoints, miss small variants|
|NGS| call smaller variants, high sensitivity, acurate sequence-level breakpoint resolution|biased towards deletions|

#### Q4: The authors used clustering method to combine data from different studies into merged CNVRs (Copy 
#### number variable regions). What are the two criteria for cluster filtering? And why did they do this filtering? 
* Genomewide assestment
* acurate breakpoint resolution
Because the aim of the study was to create a CNV map of the whole genome for variation that are not associated with phenotypes. They needed good quality data.

#### Q5: What are thresholds in stringency level 1, inclusive map (stringency level 2), and stringent map (stringency 
#### level 12) respectively? (The CNV map)
11.2%
2%
0.5%

#### Q6: Which percentage of the genome contributes to CNV in inclusive map and stringency map respectively? 
(Properties of the CNV map) 
9.5% in inclusive maps and 4.8% in stringent maps

#### Q7: By your intuition, which kind of genes are more variable between protein-coding genes and non-coding 
#### genes? How about their findings in this paper? (Functional impact of CNV)

non-coding genes, for there are no or mild phenotypic consequences where selction pressure can act on.
They found that the exons of non-coding genes had the highest proprotion of copy number variable sequences, higher than coding genes.

#### Q8: The authors generated a null CNV map and found genes for which at least 85% of the exons were 
#### homozygous deleted. What are the functions of these genes? And why did the authors say that they seem 
#### to be non-essential? (Homozygous deleted genes)
They seem non-essential, because they can be missing from genomes of apperently healthy individuals. These Genes have more family members with in which it is common to have
dublicated genes and are more prevalent in segmental dublications. 
These genes code for age-realted phenotypes, Rh-negative blood type, delayed AIDS from HIV-1 infection, suscepitbility to MS , autoimmune disease and placental function and fetal welleing
This list is non-overlapping with loss-of-functions genes reported in other studies.

#### Q9: If you are a medical doctor,  how do you use this map as a tool to assess the clinical importance of a 
#### CNV? (Clinical application of the CNV map part in Discussion)
Asses the clincial importance of a CNV:
* is it found in this CNV Map?
* does it overlap with medical relevant genes
