# Calling somatic short mutations, both single nucleotide and indels, using GATK4 Mutect2 and FilterMutectCalls

> Preliminary steps to get the aligned BAM sequence (not part of the tutorial)
> The reference sequence and the raw sequence are aligned. What we get is an aligned BAM file.
> From this file, we need to get rid of duplicates, so the duplicates are marked and the sequences are realigned.
> In the end we have the realigned proper BAM file.



matched normal = a normal reference to the tumor genome, normally fom the same person

PON = panel normal, an accumulation to matched normals, its not necessarily from the same individual
general pop = identifiy germline variation in the whole population.

Goal: to get as many tumor specific variations as possible, with not having the general variation (so one has to reduce the noise)

Reduce cross-contamination by comparing tumor normal and germline variants

Filtering = removing artefacts that occured in the process

Assign possible function to the variants

In the end you have the variants

Tutorial
Example data are based on a breast cancer cell line and its matched normal cell line derived from blood. Both cell lines are consented and known as HCC1143 and HCC1143_BL, respectively and are 2x76 paired end whole exome sequences aligned to GRCh38

![image](https://user-images.githubusercontent.com/91005577/136163533-ac818733-2317-4522-bc55-11cddd29942e.png)


## 2 Call Somatic SNV and Indels with mutect2
### 2.1 Call somatic SNVs and indels and generate a BAMOUT
Calling a somatic short mutation on HCC1143 tumor sample and matched normal using Mutect2. This command produces:

1. A raw unfiltered somatic callset restricted to the specified intervals list
2. A BAM containing reassembled alignments
3. Mutect stats file named

> What is the value of using a matched normal control?
germline variations need to be filtered out, because they are not relevant for tumor mutations.
One would find a lot more variation, specific to the individual. It would be very hard to find real tumor variants, with all the noise of other germline variants.
Mutect2 uses the matched normal to additionally exclude rare germline variation not captured by the germline resource and individual-spesific artefacts
Mutect2 uses a germline population resource towards evidence of alleles being germline.
A panel of normals (PoN) has a vital role that fills a gap between the matched normal and the population resource. It is used to catch additional sites of noise in sequencing data, e.g. mapping artefacts. Important for reducing noise.

### 2.2 Make a panel of normals (PoN)
1. Run Mutect2 in tumor-only mode on each normal BAM individually
2. Create a GenomicsDB from the normal Mutect2 calls
3. and the combine the normal calls using CreateSomaticPanelofNormals

## 3 Filter for confident somatic calls
 The goal in this step is to use filtering tools to identify which mutation candidates are likely to be real somatic mutations (compared to germline mutations of the individual having the tumor).

### 3.1 Estimate cross-sample contamination
This step is done with two tools `GetPileupSummaries` and `CalculateContamination`. We need to have only the one DNA and not other DNA.

**Run `GetPileupSummaries` to summarize read support for a set number of known variant sites.**
The tool tabulates read counts that support REF, ALT and OTHER alleles for the sites in the resource. The goal is to limit analysis to sites that are commonly variant.
One uses a common population germline resource containing only common biallelic variants.

The output is a six-column table:
> ">METADATA<" SAMPLE=HCC1143_tumor
contig	position	ref_count	alt_count	other_alt_count	allele_frequency
chr6	29942512	9	0	0	0.063
chr6	29942517	13	1	0	0.062
chr6	29942525	13	7	0	0.063
chr6	29942547	36	0	0	0.077
chr6	29942563	19	23	0	0.086
chr6	29942594	36	7	0	0.118
chr6	29942625	31	0	0	0.125
chr6	29942642	22	3	0	0.114 <
  
`alt_count` is the count of reads that support the ALT allele in the germline resource.
`allele_frecuency` corresponds to tat given in the germline resource.

The tool considers homozygous-variant sites in the sample where the alternate allele frequency (AF) in the population resource ranges between 0.01 and 0.2. This range is adjustable. We can expect a lot of contamination by alternate alleles at sites where the alternate AF is large, so those sites wouldnt tell us much. Conversely, at homozygous-alternate sites where the variant allele is rare in the population, we are more likely to observe the presence of REF or other alleles if there was cross-sample contamination, and therefore we will be able to measure contamination more accurately.

**Estimate contamination with CalculateContamination.**

For our small tumor BAM file, you can see the contamination is ~0.0191 with an error of ~0.0022. We get a slightly lower number, ~0.0120 +/– 0.00454 for the matched estimate. For the full BAM file, we see a slightly larger contamination number. This threshold informs you to be wary of calls with less than that number for the alternate allele fraction.

The resulting files from the two variations each give the fraction contamination. For our small tumor BAM file, you can see the contamination is ~0.0191 with an error of ~0.0022. We get a slightly lower number, ~0.0120 +/– 0.00454 for the matched estimate. For the full BAM file, we see a slightly larger contamination number. This threshold informs you to be wary of calls with less than that number for the alternate allele fraction.

### 3.2 Apply filters with `FilterMutectCalls`
Goal: Use the annotations within the callset, and if provided, use the contamination table to filter out variants.
For our small tumor BAM file, you can see the contamination is ~0.0191 with an error of ~0.0022. We get a slightly lower number, ~0.0120 +/– 0.00454 for the matched estimate. For the full BAM file, we see a slightly larger contamination number. This threshold informs you to be wary of calls with less than that number for the alternate allele fraction.

## 4 Review calls with IGV
Deriving a good somatic callset involves comparing callsets from different callers, manually reviewing passing and filtered calls and, if necessary, additional filtering. Manual review extends from deciphering call record annotations to the nitty-gritty of reviewing read alignments using a visualizer.

How do we account for variant calls based on the read data? Remember Mutect2 reassembles reads just like HaplotypeCaller, so the clean alignments will not necessarily reflect the calls. We must examine the BAMOUT that Mutect2's graph-assembly produces. We already generated this BAMOUT in section 1.1 (sandbox/2_tumor_normal_m2.bam). We are going to copy it into our bucket for loading into the IGV.

Navigate to the location of the genome where variants were called
We navigated IGV to the TP53 locus at chr17:7,666,402-7,689,550.

![image](https://user-images.githubusercontent.com/91005577/136170489-30f7f460-be87-4c6a-8283-d65ae8478dea.png)

**We see a C --> T variant light up in red for the tumor but not the normal. What do you think is happening in 2_tumor_normal_m2.bam?**

We see two groups, normal and tumor, whose sequences are aligned to the reference genome. for specific regions you see the reads. The reads are now colour-coded to possible haplotypes.

**What does the coverage tell you?**
The coverage displays which parts of the sequence is covered to which extend with reads. So a lower bar indicates lower coverage with reads, higher bars show a lot of coverage, with many reads covering this area.

**What are the three grouped tracks for the bamout? What does the colors indicate? What differentiates the pastel versus the gray reads?**

The firt track is a legend track showing 4 different haplotypes for this area.
The haplotypes inclued two loci, one with a mutation which shows up a lot in the tumor. The gray reads are reads which could not be annotated with a haplotype, because they are either broken or too short.


**How do you feel about this somatic call?**

# 5 Annotate mutations with FUNCOTAROR

Examine the annotations for the TP53 mutation that we views earlier in IGV.

![image](https://user-images.githubusercontent.com/91005577/136189298-eac23fad-b9fa-42f1-8ecb-229854644a4c.png)

The C/T mutation leads to a change in the aminoacid chain.

## 5.1 Review filtered indels to study the logic behind different filters

artifact_in_normal: The mutation is also in the normal cells
contamination: The mutation is not found in at least 50% of the reads
panel_of_normals:
str_contraction:
clustered_events:
germline_risk:
multiallelic: more than one mutation is special in the tumor




