# File Formats


## Whih genomic file formats exist & what are thei use cases?

### VCF Files

VCF stands for __Variant Call Format__. This text format is used for storing sequence variants (e.g. SNPs).
It consists of:
* meta information lines (starting with '##'): 
  * a 'fileformat' filed is always required. It gives information on the version of the VCF format
  * Informationlines about the INFO, FILTER and FORMAT entries used in the body of the VCF file 
  * Infomration on Date, source, reference sequence, etc.
* header line: Each header represents one column in the VCF. There are 8 fixed, mandatory columns:
  * #CHROM : identifier from the reference genome
  * POS: reference position, with the 1st base having position 1
  * ID: semi-colon separated list of unique identifiers where available
  * REF: reference base(s)
  * ALT: comma separated list of alternate non-reference alleles called on at least one of the samples
  * QUAL: phred-scaled quality score for the assertion made in ALT
  * FILTER: PASS if this position has passed all filters, i.e. a call is made at this position. Otherwise, if the site has not passed all filters, a semicolon-separated list of codes for filters that fai
  * INFO: additional information
* data line: Each of these lines contains information (see header) about a position in the genome.

The advantage using the VCF format is that only variations have to be stored together with a reference sequence. 
Therefore, it is one of the most commonly used file format when studying genetic variants.


###SAM & BAM
#### SAM
SAM stands for __Sequence Alignment Map__ and is a tab-delimited text format. 
It is commonly used to store nucleotide sequences, generated by next generation sequencing technologies.
Most often it is generated as a human readable version of its sister BAM format.
The format can store short as well as long reads, making it more versatile.

SAM files consists of:
* header section: must be prior to alignments and starts with @
* alignment section: contains 11 mandatoory fields s for
essential alignment information:
  * QNAME: Query template name. Reads/segments having identical QNAME are regarded to come from
the same template
  * FLAG: combination of bitwise flags &#8594; detailed info: https://samtools.github.io/hts-specs/SAMv1.pdf
  * RNAME: Reference sequence name of the alignment
  * POS: 1-based leftmost mapping Position of the first CIGAR operation that “consumes” a reference
base  &#8594; detailed info: https://samtools.github.io/hts-specs/SAMv1.pdf
  * MAPQ: Mapping Quality
  * CIGAR: CIGAR string  &#8594;  detailed info: https://samtools.github.io/hts-specs/SAMv1.pdf
  * RNEXT:  Reference sequence name of the primary alignment of the next read in the template
  * PNEXT:  1-based Position of the primary alignment of the next read in the template
  * TLEN:  signed observed Template length
  * SEQ: segment sequence
  * QUAL: Quality score, phred-scaled
  
 ####BAM
 BAM stands for __Binary Alignment Map__. It contains the same information as a SAM file but as binary infomration.
 Due to its binary fomrat it is used when working on the file on computer while the SAM format is there to make the file readable for us.
 
