# **Resources for genome analysis and genomic research**

&nbsp;
&nbsp;

## Cancer specific resources:

* ### [Cancer Genome Interpreter](https://www.cancergenomeinterpreter.org/home)

The Cancer Genome interpreter is a tool used primarily for the identification of oncogenic alterations in a known cancer genome.
In order to identify an alteration (point mutation, block substitution, indel, complex indel, etc.) specific input formats have to be used.
Additionally, the match can be attempted to a specific cancer type or a generic cancer type should the specific type be unknown. 
Two reference genomes (hg37 and hg19) can be used.

Upon succesful identification of an oncogenic alteration, potentially effective prescriptions are directly displayed: 

<img src="https://github.com/compbiozurich/UZH-BIO392/blob/master/course-results/2021/denis-adamec/images/CGI_Result.png" width="750">


* ### [Clinical Interpretation of Variants in Cancer](https://civicdb.org/home)

Another resource which can be used to interpret identified alterations in cancer genomes. Approximately 455 genes and a total of 2854 variants are catalogued in this database.


* ### [Cancer Driver Log](https://candl.osu.edu/)

Cancer driver log is similar in functionality to the first two resources, however entries are classified into four levels of evidence:


1. Alteration has matching FDA approved or NCCN recommended therapy.
2. Alteration has matching therapy based on evidence from clinical trials, case reports, or exceptional responders.
3. Alteration predicts for response or resistance to therapy based on evidence from pre-clinical data (in vitro or in vivo models).
4. Alteration is a putative oncogenic driver based on functional activation of a pathway.


This allows for easy decisions on the potential of different suggested prescriptions.

&nbsp;
&nbsp;

## Resources for matching sequences to known genomes

