## Why is important to construct a CNV map on healthy individuals of various ethnicities?

Answer: Because clinicians and researchers need to discriminate between benign variants, which occur in primarily healthy individuals, and pathogenic, high-risk variants, which then lead to diseases or other forms of disabilities.

## What is the CNV size that the authors defined? 

Answer: Currently, it's 50bp to 3 Mb. 

## What are the primary approaches used for CNV detection? And what are the advantages and limitations 
## for these technologies? 

Answer: Microarrays and Next-Generation-Sequencing are the most commong approaches for CNV detection. Microarrays with large clones have been used in the past, but now CGH-arrays [comparative genomic hybridization arrays] and SNP-based arrays are being used. 
Arrays are very useful for quantitative variants, but miss small variants. Duplications are more likely to be missed by SNP-based arrays and NGS-methods, but not by array CGH. NGS can detect smaller variants, has high sensitivity and high accuracy, while array-based approaches have lower resolution. Small differences in copy numbers are more likely to be caught on by CGH-array-based methods.

## The authors used clustering method to combine data from different studies into merged CNVRs (Copy 
## number variable regions). What are the two criteria for cluster filtering? And why did they do this filtering?

Answer:
