## Questions about paper "A copy number variation map of the human genome" [2015] 

# Why is important to construct a CNV map on healthy individuals of various ethnicities?

Answer: Because clinicians and researchers need to discriminate between benign variants, which occur in primarily healthy individuals, and pathogenic, high-risk variants, which then lead to diseases or other forms of disabilities.

# What is the CNV size that the authors defined? 

Answer: Currently, it's 50bp to 3 Mb. 

# What are the primary approaches used for CNV detection? And what are the advantages and limitations 
# for these technologies? 

Answer: Microarrays and Next-Generation-Sequencing are the most commong approaches for CNV detection. Microarrays with large clones have been used in the past, but now CGH-arrays [comparative genomic hybridization arrays] and SNP-based arrays are being used. 
Arrays are very useful for quantitative variants, but miss small variants. Duplications are more likely to be missed by SNP-based arrays and NGS-methods, but not by array CGH. NGS can detect smaller variants, has high sensitivity and high accuracy, while array-based approaches have lower resolution. Small differences in copy numbers are more likely to be caught on by CGH-array-based methods.

# The authors used clustering method to combine data from different studies into merged CNVRs (Copy 
# number variable regions). What are the two criteria for cluster filtering? And why did they do this filtering?

Answer: There needed to be an at least 50% reciprocal overlap, because different studies used different CNV boundaries. Then Clusters were filtered on the basis of the number of distinct subjects which carried the variant. And also filtering happened according to the number of studies with at least one variant in the cluster. In this way, the authors arrived at a "consensus CNV" data bank, which was consistent and which could be used in further analysis. 

# What are thresholds in stringency level 1, inclusive map (stringency level 2), and stringent map (stringency 
# level 12) respectively? 

Answer: Stringency level 1: Map supported by at least one subject and one study.
        Inclusive map (Stringency level 2): Map supported by at least two subjects and one study for each variant.
        Stringent map (Stringency level 12): Map supported by at least two subjects and two studies for each variant.

# Which percentage of the genome contributes to CNV in inclusive map and stringency map respectively? 

Answer: Inclusive map: 9.5% of the human genome.
        Stringent map: 4.8% of the human genome.
        
# By your intuition, which kind of genes are more variable between protein-coding genes and non-coding 
# genes? How about their findings in this paper? 

Answer: I would have thought that mostly Exons are variable in protein-coding genes, while Introns are more variable in non-coding genes. Also that genes, which are responsible for diseases are very variable.
In the paper they find, that the Exons of non-coding genes contributed the most variability. Also, many gene sets associated with disease show very little variability, contrary to my expectations. 

# The authors generated a null CNV map and found genes for which at least 85% of the exons were 
# homozygous deleted. What are the functions of these genes? And why did the authors say that they seem 
# to be non-essential?

Answer: They seem to be non-essential because they were seemingly missing from healthy individuals. They may be related to late-stage offset of certain phenotypes associated with diseases, among them age-related phenotypes, delayed AIDS from HIV-1 infection, autoimmune disease, placental function and fetal wellbeing etc. 

# If you are a medical doctor,  how do you use this map as a tool to assess the clinical importance of a 
# CNV?

Answer: Are found CNVs present in the CNV map? Are they overlapping with medically relevant genes? Comparing inclusive and stringent map to detect certain age-related phenotypes and their mechanisms and causes in the genome.
