A CNV map of healthy individuals of various ethnicities can serve as a control when examining CNV in relationship to disease.

The authors define CNVs as larger than 50bp.

Microarrays and NGS are the primary approaches used for CNV detection. Sequencing: high sensitivity, high breakpoint resolution. Array: limited resolution capacity, good for detecting duplications, more sensitive for detecting small differences in copy number.

2 criteria: number of distinct subjects that carry the variant and number of studies with at least one variant in the cluster. Exclude singletons and study-specific artefacts.

Level 1: at least 1 subject, 1 study (includes everything). Level 2: at least 2 subjects and 1 study. Level 12: at least 2 subjects and 2 studies.

Percentage of genome contributing to CNVs by map: inclusive: 9.5%, stringent: 4.8%

Exons of all genes were more variable than the genome average (background), exons of non-coding enes had the highest proportion of CNV sequence. Exons of genes associated with disease or other fitness-altering phenotypes were less variable than the genome average.

Their function could be redudant (paralogues) or related to late-onset phenotypic consequences that do not substantially reduce fitness. They are considered non-essential, because they could be missing from the genomes of apparently healthy individuals.

Compare CNV found in case subject to CNV map and medically relevant genes. A CNV in a medically relevant gene, which is not on the CNV map suggests medical importance.
