# Classes of structural variation

* Deletion (> 50 bp)
* Novel sequence insertion (> 50 bp)
* Mobile-element insertion
* Tandem duplication
* Interspersed duplication
* Inversion
* Translocation
See ![here](https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fnrg2958/MediaObjects/41576_2011_Article_BFnrg2958_Fig1_HTML.jpg)

# Discovery and genotyping of structural variations using arrays


* Array Comparative Genotype Hybridization (aCGH)
Technology:

1) Test and Control DNA are fluorescently labeled using different colors.
2) In parallel, probes complementary to specific DNA sites are designed which are long oligonucleotides or historically BAC clones and placed on glass slides.
3) For each array, both test and control samples are added. Outcome: if green + red --> two copies for each (color is yellow), copy loss (red), copy gain (green).
And so on for multiple probes. For example, Roche NimbleGen and Agilent Technologies have 2.1M and 1M oligonucleotide probe kits (50-75-mers), respectively. To detect CNV, 3 to 10 sequential probes must \
indicate the copy number alteration. Today, using custom, high-probe density arrays such as 24M and 42M aCGH, allows to detect 500bp CNV and their breakpoints.

Advantages: ultra-high resolution arrays can distinguish between small structural variations (<500 bp).
Disadvantages: ultra-high resolution arrays can be carried out on relatively low number of samples which unables to characterize the reference samples very precisely.


* SNP array:
Technology:

1) Oligonucleotide probes are designed by single-base extension or differential hybridization method per specific SNP annotated across DNA differences
2) Each array is carried out on single sample by labeling DNA with fluorescents.
3) Then, log-transformed ratios are generated by clustering intensities across many samples.

Advantages: can detect uniparental disomy due to SNP-specific parental allele through calculation of B-allele frequency metric.
Disadvantages: lower signal-to-noise ratio to aCGH.

Today, resolution of SNP arrays has increased and SNP-targeting selection has furthered. Moreover, average heterozygosity is inversely proportional with the required number of SNP probes to detect \
uniparental disomy because under low heterozygosity, frequency of different alleles is low and more probes needed to detect variant allele.

Limitations of Arrays:

- Different probe density and per-rpobe speicificity which has induced creation of multiple algortihms with a lot of parameter optimization.

- Assay sensitivity to detect short CNV and assumptions like diploid reference genome even if it is not the case. For example: duplication in reference genome causes more than two copies of one region.

See ![aCGH method](https://www.researchgate.net/profile/Sebastiano_Cavallaro/publication/305078193/figure/fig2/AS:391208695287857@1470282829808/Array-CGH-procedure-is-characterized-by-the-isolation-of-DN$
Advantages of Arrays: cost and throughput

* Single Molecule Analysis:

Immobilized genome is digested using restriction enzymes and based on ordering of portions is compared to in silico digested reference genome.This method allows to detect inversions, translocations, \
copy number alterations and their locations.

# Sequencing-based computational approach

- NGS technologies seem to surpass arrays in the near future for CNV detection but are presented with computational and bioinformatic challenges. A lot of the computational algorithms have been \
designed to investigate capillary sequencing reads and fully-sequenced large-insert clones.

- Methods for computational analysis of CNVs through NGS technologies such whole-exome-sequencing and whole-genome-sequencing:

* Read depth: method uses the deviation from Poisson distribution in mapping depth which at high level is an indicator of duplication and at low level of a deletion. But not all structural \
variations can be identified using this technique. To discover smaller duplications and deletions at smaller breakpoint resolution **event-wise-testing algorithm** and **CVNator**  are used.

* Read-pair technologies:

- This can detect all types of strucutural variations. For instance, if paired-end reads are aligned further apart on the reference genome, it is deletion. If aligned closer on the reference genome, \
it is insertion. If orientation of the reads are switched, then inversion or tandem duplication. Computational tools that use read-pair approach: PEMer, VariationHunter, BreakDancer, MoDIL, MoGUL, \
HYDRA, Corona and SPANNER.

* Split-read approach: approach uses the space between reads aligned to the reference genome. See the image below. Approach can detect all types of structural variations. And Pindel algorithm is an \
example.


* Assembly-based approach: assembly of reads into contigs are done through de novo assembly and local assembly techniques. Well-known de novo assembly algorithms for next-generation whole-genome \
shotgun (NG-WGS) data include EULER-USR, ABySS, SOAPdenovo and ALLPATHS-LG. Using the Cortex assembler, variant assembly can be done entirely de novo or with different degrees of information from a \
reference, and Cortex has the ability to simultaneously assemble multiple genomes and call SVs between samples without the need for a reference. The NovelSeq framework merges de novo and local-assembly \
methods to characterize novel sequence insertions and, finally, TIGRA aims to improve breakpoint estimations in SV discovery. Assembly-based approach can identify all types of structural variations.

See ![computational approach to find structural variants](https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fnrg2958/MediaObjects/41576_2011_Article_BFnrg2958_Fig2_HTML.jpg)



Softwares:

- SPANNER, CNVer and Genome STRiP combine read-pair and read-depth methods in different ways to more reliably detect CNVs.
- BreakSeq was developed to build a library of SV breakpoints discovered in the literature

