# Task 4

## for storing called variants:

- SAM (Sequence Alignment/Map) 
	- Text file format for storing data
	- Optional header section followed by an alignment section  
	- Nucleotide sequences (generated by next generation sequencing technologies) and unmapped sequences
	- Same as BAM, but more readable for humans 
	- [SAM](https://samtools.github.io/hts-specs/SAMv1.pdf) 
            

- VCF (Variant Call Format) 
	- Text file format (often compressed) 
	- Information about position in genome 
	- [VCF](https://samtools.github.io/hts-specs/VCFv4.1.pdf) |


- FASTA 
	- Text-based format for nucleotide or amino acid sequences (represented by single-letter codons) 
	- Single-line description followed by lines of sequence data 
	- Text lines should be in length shorter than 80 characters 
	- [FASTA](https://en.wikipedia.org/wiki/FASTA_format) 
	- example:
		![image_FASTA](https://www.slideshare.net/sesejun/20110524zurichngs-1st-pub-8133028)


## for full archival purposes:

- BAM (Binary Alignment/Map)
	- Binary format version of SAM 
	- Used to store information, not readable for humans 
	- [BAM](https://genome.sph.umich.edu/wiki/BAM)



## for browser visualization:

- BED (Browser Extensible Data) 
	- Text file format for storing coordinates and associated annotations 
	- Representation of data by columns separated by spaces and tabs
	- [BED](https://en.wikipedia.org/wiki/BED_(file_format))
