# Task 4 - WGS, WES, Cost Estimations and File Formats


## Whole Genome Sequencing (WGS) and Whole Exome Sequencing (WES)
	- WGS is an attempt to sequence the entiretly genome (95% to 98% of the genome can be sequenced) and WES focuses on the protein coding sequences (about 2% of the genome)
	- The decision which sequencing should be used depends what is wanted to be known about the organism
	- Both are for detection of pathogenic genetic variations such as single nucleotide polymorphisms (SNPs) and insertions and/or deletions (Indels)
	- Sources: 
		- [AllSeq](https://allseq.com/knowledge-bank/1000-genome/wgs-vs-wes/)
		- [CCI](https://www.uniklinik-freiburg.de/cci/forschung/bodo-grimbacher/forschungsgebiete/wes-wgs.html)



## Estimation of costs

	- Due to the new machine technologies for sequencing the costs for WGS and WES dropped immensively to appr. 1,000 US Dollars ([NIH](https://www.genome.gov/about-genomics/fact-sheets/Sequencing-Human-Genome-cost))
	- The problem nowadays are the costs that come along for storing the data	
		- Associated costs:
			- Cost factors are employees, technology, storage costs and so on
				- Example: Raw Storage costs for BAM format are appr. 40CHF/genome ([Slides](https://compbiozurich.org/UZH-BIO392/course-material/2020/2020-09-18-BIO392-files.pdf))


## File Formats

- **SAM (Sequence Alignment/Map)** 
	- Generic alignment in text file format for storing read alignments against reference sequences
	- Supports short and long reads (up to 128 Mbp) by different sequencing platforms
	- Format: TAB-delimited text format with optional header section followed by an alignment section of nucleotide sequences (generated by next generation sequencing technologies) and unmapped sequences
	- Same as BAM, since it is not binary, it is more readable for humans
	- [Example](https://sites.google.com/site/wiki4metagenomics/tools/samtools/bam-sam-file-format):
		
		>@HD    VN:1.3    SO:coordinate
		@SQ    SN:conticA    LN:443
		readID43GYAX15:7:1:1202:19894/1    256    contig43    613960    1    65M    *    0    
		0    CCAGCGCGAACGAAATCCGCATGCGTCTGGTCGTTGCACGGAACGGCGGCGGTGTGATGCACGGC    EDDEEDEE=EE?
		DE??DDDBADEBEFFFDBEFFEBCBC=?BEEEE@=:?::?7?:8-6?7?@??#    AS:i:0    XS:i:0  XN:i:0  
		XM:i:0  XO:i:0  XG:i:0  NM:i:0  MD:Z:65  YT:Z:UU
		
	- Sources:
		- [Paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2723002/)
		- [GitHub](https://samtools.github.io/hts-specs/SAMv1.pdf)
		- [Metagenomics](https://sites.google.com/site/wiki4metagenomics/tools/samtools/bam-sam-file-format) 	



- **BAM (Binary Alignment/Map)**
	- Format: Binary format version of SAM 
	- Used to store information, not readable for humans 
	- Source:
		- [Center for Statistical Genetics](https://genome.sph.umich.edu/wiki/BAM)



- **CRAM**
	- Compressed columnar file format for storing biological sequences, which are aligned to a reference sequence
	- CRAM files are alignment files such like BAM files, but CRAM was designed to reduce the disk footprint of alignment data even more
	- Sources:
		- [Wikipedia](https://en.wikipedia.org/wiki/CRAM_(file_format))
		- [IGSR](https://www.internationalgenome.org/faq/what-are-cram-files/)



- **VCF (Variant Call Format)** 
	- A generic format of variants for storing DNA polymorphism data (like SNPs, deletions and structural variants) as well with rich annotations
	- Text file format (often compressed) 
	- Format: Meta-information lines, a header line as well data lines that contain information about position in genome 
	- [Example](https://samtools.github.io/hts-specs/VCFv4.2.pdf):
		
		>##fileformat=VCFv4.2
		##fileDate=20090805
		##source=myImputationProgramV3.1
		##reference=file:///seq/references/1000GenomesPilot-NCBI36.fasta
		##contig=<ID=20,length=62435964,assembly=B36,md5=f126cdf8a6e 0c7f379d618ff66beb2da,species="Homo sapiens",taxonomy=x>
		##phasing=partial
		##INFO=<ID=NS,Number=1,Type=Integer,Description="Number of Samples With Data">
		##INFO=<ID=DP,Number=1,Type=Integer,Description="Total Depth">
		##INFO=<ID=AF,Number=A,Type=Float,Description="Allele Frequency">
		##INFO=<ID=AA,Number=1,Type=String,Description="Ancestral Allele">
		##INFO=<ID=DB,Number=0,Type=Flag,Description="dbSNP membership, build 129">
		##INFO=<ID=H2,Number=0,Type=Flag,Description="HapMap2 membership">
		##FILTER=<ID=q10,Description="Quality below 10">
		##FILTER=<ID=s50,Description="Less than 50% of samples have data">
		##FORMAT=<ID=GT,Number=1,Type=String,Description="Genotype">
		##FORMAT=<ID=GQ,Number=1,Type=Integer,Description="Genotype Quality">
		##FORMAT=<ID=DP,Number=1,Type=Integer,Description="Read Depth">
		##FORMAT=<ID=HQ,Number=2,Type=Integer,Description="Haplotype Quality">
		#CHROM POS ID REF ALT QUAL FILTER INFO FORMAT NA00001 NA00002 NA00003
		20 14370 rs6054257 G A 29 PASS NS=3;DP=14;AF=0.5;DB;H2 GT:GQ:DP:HQ 0|0:48:1:51,51 1|0:48:8:51,51 1/1:43:5:.,.
		20 17330 . T A 3 q10 NS=3;DP=11;AF=0.017 GT:GQ:DP:HQ 0|0:49:3:58,50 0|1:3:5:65,3 0/0:41:3
		20 1110696 rs6040355 A G,T 67 PASS NS=2;DP=10;AF=0.333,0.667;AA=T;DB GT:GQ:DP:HQ 1|2:21:6:23,27 2|1:2:0:18,2 2/2:35:4
		20 1230237 . T . 47 PASS NS=3;DP=13;AA=T GT:GQ:DP:HQ 0|0:54:7:56,60 0|0:48:4:51,51 0/0:61:2
		20 1234567 microsat1 GTC G,GTCT 50 PASS NS=3;DP=9;AA=G GT:GQ:DP 0/1:35:4 0/2:17:2 1/1:40:3
		
	- Sources:
		- [Paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3137218/)
		- [GitHub](https://samtools.github.io/hts-specs/VCFv4.1.pdf)
		- [Lecture slides](https://compbiozurich.org/UZH-BIO392/course-material/2020/2020-09-18-BIO392-files.pdf)


- **FASTA** 
	- Text-based format for nucleotide or amino acid sequences (represented by single-letter codons) 
	- Fromat: Single-line description (distinguished by ">" symbol) followed by lines of sequence data 
	- Text lines should be in length shorter than 80 characters 
	- [Example](https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&PAGE_TYPE=BlastDocs&DOC_TYPE=BlastHelp)
		
		> .>P01013 GENE X PROTEIN (OVALBUMIN-RELATED)
		QIKDLLVSSSTDLDTTLVLVNAIYFKGMWKTAFNAEDTREMPFHVTKQESKPVQMMCMNNSFNVATLPAE
		KMKILELPFASGDLSMLVLLPDEVSDLERIEKTINFEKLTEWTNPNTMEKRRVKVYLPQMKIEEKYNLTS
		VLMALGMTDLFIPSANLTGISSAESLKISQAVHGAFMELSEDGIEMAGSTGVIEDIKHSPESEQFRADHP
		FLFLIKHNPTNTIVYFGRYWSP
		
	- Sources: 
		- [Wikipedia](https://en.wikipedia.org/wiki/FASTA_format) 
		- [NCBI](https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&PAGE_TYPE=BlastDocs&DOC_TYPE=BlastHelp)



- **BED (Browser Extensible Data)** 
	- Text file format for storing coordinates and associated annotations 
	- Fromat: Representation of data by columns separated by spaces and tabs
	- [Example](https://m.ensembl.org/info/website/upload/bed.html):
		
		>chr1  213941196  213942363 <br>
		chr1  213942363  213943530 <br>
		chr1  213943530  213944697 <br>
		chr2  158364697  158365864
		
	- Sources:
		- [Wikipedia](https://en.wikipedia.org/wiki/BED_(file_format))
		- [Ensembl](https://m.ensembl.org/info/website/upload/bed.html)



- **MPEG-G (Moving Picture Experts Group for Genomic information)**
	- Utilizes latest technology to further compress, store, transmit and process sequencing data for complex use, where the amount of information required is reduced to loslessly store and transmit DNA reads
	- The complex use possibilities are data streaming, genomic studies aggregation or annotation and linkage of genomic segments.
	- Example of file creation:
		1. Place the reads in input file (corresponding to the reference genome)
		2. Classify the reads in each bin according to the 6 classes (like P = perfect match with reference genome)
		3. Convert the reads of each bin to a subset of 18 descriptors (specific to the defined class)
		4. Create a matrix where the descriptors are in the columns
		5. Compress each descriptor column
		6. The compressed descriptors of a class of a bin are compressed in a Access Unit (AU), where per bin are up to 6 AUs
		
		![Picture of worflow](https://blog.chiariglione.org/wp-content/uploads/2018/12/Untitled-8.png)
 
	- Sources:
		- [MPEG-G](https://mpeg-g.org/about/)
		- [Blog](https://blog.chiariglione.org/still-more-to-say-about-mpeg-standards/)