# Task 4

for storing called variants:

file format | format and functions | source |
--- | --- | --- |

SAM (Sequence Alignment/Map) | - Optional header section followed by an alignment section  <br> - Storing data: nucleotide sequences (generated by next generation sequencing technologies) and unmapped sequences <br> - same as BAM, but more readable for humans | [SAM](https://samtools.github.io/hts-specs/SAMv1.pdf) |
                               
VCF (Variant Call Format) | - text file format (oftem compressed) <br> - information about position in genome | [VCF](https://samtools.github.io/hts-specs/VCFv4.1.pdf) |

FASTA | - text-based format for nucleotide or amino acid sequences (represented by single-letter codons)- single-line description followed by lines of sequence data <br> - text lines should be in length shorter than 80 characters | [FASTA](https://en.wikipedia.org/wiki/FASTA_format) |


for full archival purposes:

file format | format and functions | source |
--- | --- | --- |
BAM (Binary Alignment/Map) | - binary format of SAM <br> - used to store information, not readable for humans | [BAM](https://genome.sph.umich.edu/wiki/BAM)



for browser visualization:

file format | format and functions | source |
--- | --- | --- |
BED (Browser Extensible Data) | - text file format for storing coordinates and associated annotations <br> - representation of data by columns separated by spaces and tabs | [BED](https://en.wikipedia.org/wiki/BED_(file_format))
