# Task 4 - File Formats

Task description:
Please provide 1-page size estimates and reasoning for the use of the different file types (i.e. which would you use for storing called variants, which for full archival purposes, browser visualisation), for 3-5 formats.
Submit your files (.md) per pull request to your Github directory.


## ... for storing called variants:

- **SAM (Sequence Alignment/Map)** 
	- Generic alignment in text file format for storing read alignments against reference sequences
	- Supports short and long reads (up to 128 Mbp) by different sequencing platforms
	- Format: TAB-delimited text format with optional header section followed by an alignment section of nucleotide sequences (generated by next generation sequencing technologies) and unmapped sequences
	- Same as BAM, since it is not binary, it is more readable for humans
	- [Example](https://sites.google.com/site/wiki4metagenomics/tools/samtools/bam-sam-file-format):
		>
		@HD    VN:1.3    SO:coordinate
		@SQ    SN:conticA    LN:443
		readID43GYAX15:7:1:1202:19894/1    256    contig43    613960    1    65M    *    0    
		0    CCAGCGCGAACGAAATCCGCATGCGTCTGGTCGTTGCACGGAACGGCGGCGGTGTGATGCACGGC    EDDEEDEE=EE?
		DE??DDDBADEBEFFFDBEFFEBCBC=?BEEEE@=:?::?7?:8-6?7?@??#    AS:i:0    XS:i:0  XN:i:0  
		XM:i:0  XO:i:0  XG:i:0  NM:i:0  MD:Z:65  YT:Z:UU
		
	- Sources:
		- [Paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2723002/)
		- [GitHub](https://samtools.github.io/hts-specs/SAMv1.pdf)
		- [Metagenomics](https://sites.google.com/site/wiki4metagenomics/tools/samtools/bam-sam-file-format) 	

            

- **VCF (Variant Call Format)** 
	- A generic format for storing DNA polymorphism data (like SNPs, deletions and structural variants) as well with rich annotations
	- Text file format (often compressed) 
	- Format: Meta-information lines, a header line as well data lines that contain information about position in genome 
	- [Example](https://samtools.github.io/hts-specs/VCFv4.2.pdf):
		>
		##fileformat=VCFv4.2
		##fileDate=20090805
		##source=myImputationProgramV3.1
		##reference=file:///seq/references/1000GenomesPilot-NCBI36.fasta
		##contig=<ID=20,length=62435964,assembly=B36,md5=f126cdf8a6e 0c7f379d618ff66beb2da,species="Homo sapiens",taxonomy=x>
		##phasing=partial
		##INFO=<ID=NS,Number=1,Type=Integer,Description="Number of Samples With Data">
		##INFO=<ID=DP,Number=1,Type=Integer,Description="Total Depth">
		##INFO=<ID=AF,Number=A,Type=Float,Description="Allele Frequency">
		##INFO=<ID=AA,Number=1,Type=String,Description="Ancestral Allele">
		##INFO=<ID=DB,Number=0,Type=Flag,Description="dbSNP membership, build 129">
		##INFO=<ID=H2,Number=0,Type=Flag,Description="HapMap2 membership">
		##FILTER=<ID=q10,Description="Quality below 10">
		##FILTER=<ID=s50,Description="Less than 50% of samples have data">
		##FORMAT=<ID=GT,Number=1,Type=String,Description="Genotype">
		##FORMAT=<ID=GQ,Number=1,Type=Integer,Description="Genotype Quality">
		##FORMAT=<ID=DP,Number=1,Type=Integer,Description="Read Depth">
		##FORMAT=<ID=HQ,Number=2,Type=Integer,Description="Haplotype Quality">
		#CHROM POS ID REF ALT QUAL FILTER INFO FORMAT NA00001 NA00002 NA00003
		20 14370 rs6054257 G A 29 PASS NS=3;DP=14;AF=0.5;DB;H2 GT:GQ:DP:HQ 0|0:48:1:51,51 1|0:48:8:51,51 1/1:43:5:.,.
		20 17330 . T A 3 q10 NS=3;DP=11;AF=0.017 GT:GQ:DP:HQ 0|0:49:3:58,50 0|1:3:5:65,3 0/0:41:3
		20 1110696 rs6040355 A G,T 67 PASS NS=2;DP=10;AF=0.333,0.667;AA=T;DB GT:GQ:DP:HQ 1|2:21:6:23,27 2|1:2:0:18,2 2/2:35:4
		20 1230237 . T . 47 PASS NS=3;DP=13;AA=T GT:GQ:DP:HQ 0|0:54:7:56,60 0|0:48:4:51,51 0/0:61:2
		20 1234567 microsat1 GTC G,GTCT 50 PASS NS=3;DP=9;AA=G GT:GQ:DP 0/1:35:4 0/2:17:2 1/1:40:3
		
	- Sources:
		- [Paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3137218/)
		- [GitHub](https://samtools.github.io/hts-specs/VCFv4.1.pdf)
		- [Lecture slides](https://compbiozurich.org/UZH-BIO392/course-material/2020/2020-09-18-BIO392-files.pdf)


- **FASTA** 
	- Text-based format for nucleotide or amino acid sequences (represented by single-letter codons) 
	- Fromat: Single-line description (distinguished by ">" symbol) followed by lines of sequence data 
	- Text lines should be in length shorter than 80 characters 
	- [Example](https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&PAGE_TYPE=BlastDocs&DOC_TYPE=BlastHelp)
		>
		>P01013 GENE X PROTEIN (OVALBUMIN-RELATED)
		QIKDLLVSSSTDLDTTLVLVNAIYFKGMWKTAFNAEDTREMPFHVTKQESKPVQMMCMNNSFNVATLPAE
		KMKILELPFASGDLSMLVLLPDEVSDLERIEKTINFEKLTEWTNPNTMEKRRVKVYLPQMKIEEKYNLTS
		VLMALGMTDLFIPSANLTGISSAESLKISQAVHGAFMELSEDGIEMAGSTGVIEDIKHSPESEQFRADHP
		FLFLIKHNPTNTIVYFGRYWSP
		
	- Sources: 
		- [Wikipedia](https://en.wikipedia.org/wiki/FASTA_format) 
		- [NCBI](https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&PAGE_TYPE=BlastDocs&DOC_TYPE=BlastHelp)


## ... for full archival purposes:

- **BAM (Binary Alignment/Map)**
	- Format: Binary format version of SAM 
	- Used to store information, not readable for humans 
	- Source:
		- [Center for Statistical Genetics](https://genome.sph.umich.edu/wiki/BAM)



## ... for browser visualization:

- **BED (Browser Extensible Data)** 
	- Text file format for storing coordinates and associated annotations 
	- Fromat: Representation of data by columns separated by spaces and tabs
	- [Example](https://m.ensembl.org/info/website/upload/bed.html):
		>
		chr1  213941196  213942363
		chr1  213942363  213943530
		chr1  213943530  213944697
		chr2  158364697  158365864
		
	- Source:
		- [Wikipedia](https://en.wikipedia.org/wiki/BED_(file_format))
		- [Ensembl](https://m.ensembl.org/info/website/upload/bed.html)
