Genome Browsers
==================

### What is a genome browser?
In bioinformatics, a genome browser is a graphical interface for display of information from a biological database for genomic data. Genome browsers enable researchers to visualize and browse entire genomes with annotated data including gene prediction and structure, proteins, expression, regulation, variation, comparative analysis, etc. Annotated data is usually from multiple diverse sources.

---
## Ensembl Genome Browser 
Ensembl aims to provide a centralized resource for geneticists, molecular biologists and other researchers studying the genomes of our own species and other vertebrates and model organisms. 

### Functions:
* Compare genes across species
* Find SNPs and other variants for specific genes (comparative genomics)
* Gene expression in different tissues
* Retrieve gene sequence

### Features:
* Comparative genomics
*	Ontologies (Information about the function oft the gene)
* Phenotypes (Phenotypes, diseases and traits associated with this gene)
* Genetic Variants
* Gene expression
* Pathway
* Regulation

### Tools (all online):
* Variant Effect Predictor (VEP)
  
  Analyse your own variants and predict the functional consequences of known and unknown variants
* BLAST/BLAT
  
  Search our genomes for your DNA or protein sequence.
* File Chameleon
 
  Convert Ensembl files for use with other analysis tools
*   Assembly Converter
    
    Map (liftover) your data's coordinates to the current assembly
* ID History Converter
  
  Convert a set of Ensembl IDs from a previous release into their current equivalents
*	Linkage Disequilibrium Calculator:
 
  Linkage Disequilibrium (LD) Calculator is a tool for calculating LD between variants using genotypes from a  selected population (Are allels inherited independently or not? Crossingover)
* VCF to PED converter
  
  Parse a vcf file to create a linkage pedigree file (ped) and a marker information file, which together may be loaded into ld visualization tools like Haploview.
*	Data Slicer
  
  Get a subset of data from a BAM or VCF file
*	Post-GWAs
  
  Upload GWAS summary statistics and highlight likely causal gene candidates

---
## UniProt
UniProt is a freely accessible database of protein sequence and functional information about the proteins.

### Features:
*	UniProtKB (Protein Knowledge Base)
*	UniParc (Contains publicly available protein sequences → only uniqe sequences get stored)
*	UniRef (Clusters sets of sequences → gives complete coverage)
*	Proteomes (UniProt provides proteomes for species with completely sequenced genomes)
*	Supporting data (Literature citations, Taxonomy, Subcellular locations, Cross-ref. Databases, Diseases, Keywords)

### Tools (online):
*	BLAST
  
  Finds regions of local similarity between sequences, which can be used to infer functional and evolutionary relationships between sequences as well as help identify members of gene families
*	Align
  
  Align two or more protein sequences to view their characteristics (identitiy, similarity etc.) alongside each other.
*	Retrieve/ID mapping
  
  Convert identifiers which are of a different type to UniProt identifiers or vice versa and download the identifier lists. Can then also be downloaded in different formates.
*	Peptide search
  
  Give a peptide sequence and the tool is searching to which species it belongs and what protein it is
