# Questions about the paper *A copy number of the human genome*

CNV: larger than 50bp

## Why is important to construct a CNV map on health individuals of various ethnicities?
Differents CNVs have different imoact in other ethnicities. Therefore it is importent to have a wide range of different population to be able to categorize each CNV correctly.

## What is the CNV size that the authors defined? 
50bp to 3Mb

## What are the primary approaches used for CNV detection? And what are the advantages and limitations for these technologies?
Primary approaches: Microarray and NSG; No single strategy can capture the entire spectrum of variation. Array basewd detection are suitable for quantitative variants. 
Duplication are more often missed in SNP-based approaches than in NGS but CGH is the most readily detector. Sequencing has a higher resolution capacity.

## The authors used clustering method to combine data from different studies into merged CNVRs (Copy number variable regions). What are the two criteria for cluster filtering? And why did they do this filtering?
The two criterias were the basis of the number of distinct subject that carrry the variant and the number of distinct studies with at lieast one vaoant in the cluster.
The filter based on the number of studies ensured the exclusion of potential study-specific artefacts.

## What are thresholds in stringency level 1, inclusive map (stringency level 2), and stringent map (stringency level 12) respectively?
At least one subject and one study; at least two subjects and one study; at least two subjects and two studies

## Which percentage of the genome contributes to CNV in inclusive map and stringency map respectively?
Stringency map: 9.5% (7.5% for gains, 3.9% for losses)
Inclusive map:1.1- 16.4% for gains and 4.3 - 19.2% for losses

## By your intuition, which kind of genes are more variable between protein-coding genes and non-coding genes? How about their findings in this paper?
My intuition: non.coding genes are more variable.
Paper: Exons of non-coding genes have highest proportion of CNVs

## The authors generated a null CNV map and found genes for which at least 85% of the exons were homozygous deleted. What are the functions of these genes? And why did the authors say that they seem to be non-essential?
They could be missing from genomes from healthy people there for they need to be non-essential. Function may be redundent or related to late-onset phenotypic consequences that do not reduce fitness.
Some of them are age-related phenotypes, Rh-neg bloos type, susceptibility to MS, autoimmune disease and placental and fetal wellbeing.

## If you are a medical doctor, how do you use this map as a tool to assess the clinical importance of a CNV?
Us it as a refference to see in what context such a CNV may relate with a disease.
